File: leetcode/c++/0048-rotate-image.cc
// https://leetcode.com/problems/rotate-image/
// Medium

#include<vector>

using namespace std;

class Solution {
public:
    void rotate(vector<vector<int>>& matrix) {
        int n = matrix.size();
        for (int i = 0; i < (n+1)/2; ++i) {
            for (int j = 0; j < n/2; ++j) {
                swap(matrix[i][j], matrix[j][n-i-1]);
                swap(matrix[i][j], matrix[n-i-1][n-j-1]);
                swap(matrix[i][j], matrix[n-j-1][i]);
            }
        }
    }
};

File: leetcode/python/0048-rotate-image.py
# https://leetcode.com/problems/rotate-image/
# Medium

class Solution:
    def rotate(self, matrix: List[List[int]]) -> None:
        w = -(len(matrix) // -2)
        l = len(matrix) // 2
        for i in range(w):
            for j in range(l):
                matrix[i][j], matrix[j][len(matrix)-i-1] = matrix[j][len(matrix)-i-1], matrix[i][j]
                matrix[i][j], matrix[len(matrix)-i-1][len(matrix)-j-1] = matrix[len(matrix)-i-1][len(matrix)-j-1], matrix[i][j]
                matrix[i][j], matrix[len(matrix)-j-1][i] = matrix[len(matrix)-j-1][i], matrix[i][j]
